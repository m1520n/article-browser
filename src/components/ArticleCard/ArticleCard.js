import React from 'react';
import PropTypes from 'prop-types';
import { format, parseISO } from 'date-fns';

import {
  ArticleCardWrapper,
  CoverImage,
  CardTitle,
  CardText,
  ReadMoreButton,
  CardGroup,
  Date,
  Author,
  Source,
} from './ArticleCard.style';

const propTypes = {
  article: PropTypes.shape({
    content: PropTypes.string,
    urlToImage: PropTypes.string,
    title: PropTypes.string,
    publishedAt: PropTypes.string,
    author: PropTypes.string,
    source: PropTypes.shape({
      name: PropTypes.string,
    }),
  }).isRequired,
};

const ArticleCard = ({ article }) => (
  <ArticleCardWrapper>
    <CoverImage alt={article.title} url={article.urlToImage} />
    <CardGroup>
      <Date>{format(parseISO(article.publishedAt), 'MMM dd yyyy')}</Date>
      <Author>{article.author}</Author>
      <Source>{article.source.name}</Source>
    </CardGroup>
    <CardTitle>{article.title}</CardTitle>
    <CardText>{article.content}</CardText>
    <ReadMoreButton block>Read More</ReadMoreButton>
  </ArticleCardWrapper>
);

ArticleCard.propTypes = propTypes;

export default ArticleCard;
