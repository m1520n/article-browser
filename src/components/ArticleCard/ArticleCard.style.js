import styled from 'styled-components';

import Button from '../Button/Button';

import { fonts, colors, breakpoints } from '../../styles/variables';

export const ArticleCardWrapper = styled.div`
  flex: 0 1 30%;
  width: 37rem;
  font-family: ${fonts.arialMT};
  margin: 2.6rem 2.7rem 1.7rem 0;

  @media ${breakpoints.tablet} {
    flex: 1;
    flex: 0 1 45%;
  }

  @media ${breakpoints.mobile} {
    flex: 1;
    widht: 100%;
  }
`;

export const CoverImage = styled.div`
  width: 100%;
  height: 21.2rem;
  background-image: url(${({ url }) => url});
  background-size: cover;
`;

export const CardTitle = styled.h2`
  font-size: 2.4rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-bottom: 1.7rem;
  margin-top: 0.8rem;
  font-weight: normal;
`;

export const CardText = styled.p`
  font-family: ${fonts.arialMT};
  font-size: 1.7rem;
  height: 7.6rem;
  overflow: hidden;
  text-overflow: ellipsis;
  color: ${colors.darkSlate};
  margin-bottom: 3.2rem;
`;

export const ReadMoreButton = styled(Button)`
  background-color: ${colors.royalBlue};
  color: ${colors.white};
  width: 100%;
  height: 4.9rem;
  margin-top: 3.2rem;
`;

export const CardGroup = styled.div`
  margin-top: 2.4rem;
`;

export const Date = styled.span`
  color: ${colors.blackish};
  font-size: 1.4rem;
  font-family: ${fonts.arialMT};
  margin-right: 1.6rem;
`;

export const Author = styled.span`
  color: ${colors.blackish};
  font-size: 1.4rem;
  margin-right: 1.6rem;
  font-family: ${fonts.arialMT};
`;

export const Source = styled.a`
  color: ${colors.royalBlue};
  font-size: 1.4rem;
  font-family: ${fonts.arialMT};

  :hover {
    color: ${colors.sapphire};
  }
`;
