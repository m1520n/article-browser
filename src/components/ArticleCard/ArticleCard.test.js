import React from 'react';
import renderer from 'react-test-renderer';

import ArticleCard from './ArticleCard';

test('ArticleCard is rendered correctly', () => {
  const article = {
    content: 'CONTENT',
    author: 'AUTHOR',
    urlToImage: 'URL',
    title: 'TITLE',
    publishedAt: '2019-08-10T12:00:00Z',
    source: {
      name: 'NAME',
    },
  };

  const component = renderer.create(<ArticleCard article={article} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
