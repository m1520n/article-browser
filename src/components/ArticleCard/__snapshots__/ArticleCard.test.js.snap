// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard is rendered correctly 1`] = `
<div
  className="sc-bwzfXH hwPgcX"
>
  <div
    alt="TITLE"
    className="sc-htpNat eLZMyj"
  />
  <div
    className="sc-bZQynM bdZCbq"
  >
    <span
      className="sc-gzVnrw lFPpP"
    >
      Aug 10 2019
    </span>
    <span
      className="sc-htoDjs bhWAGR"
    >
      AUTHOR
    </span>
    <a
      className="sc-dnqmqq JTDYf"
    >
      NAME
    </a>
  </div>
  <h2
    className="sc-bxivhb ladvwN"
  >
    TITLE
  </h2>
  <p
    className="sc-ifAKCX icdYMR"
  >
    CONTENT
  </p>
  <button
    className="sc-bdVaJa doiMwP"
    disabled={false}
    onClick={[Function]}
    size="regular"
    type="primary"
  >
    Read More
  </button>
</div>
`;
